### DISCUSSION TITLE
Exciting a 3D conductor CPW on openEMS

### DISCUSSION THREAD

**Discussion by LeoGalloGomes** [thliebig/openEMS-Project#175] [Q&A Usage] [answered]
Hi all!
I come to you with a philosophical question/discussion: how could one excite a CPW featuring thick conductors on openEMS? More specifically,
1) Could one use a waveguide port like in HFSS to excite the CPW mode on that 3 conductor geometry?
2) Could one use the CPW port described in the documentation? Does it also work for thick conductors? And should it have a 50 ohm Zc or should it be an extension of the DUT? (Context, if I were to simulate a high Zc line, should the port be a 50 ohm line or should it be an extension of that high Zc line?)
3) Lumped ports. I don't really like the "brute force" approach of adding a PEC sheet to join the two grounds, acting as the negative of the lumped port, because this adds extra series inductance to the line, disturbing the results. However that's the easiest way. Another way would be to force an "even mode" on the three conductors using 4, 100 ohm terminals and placing them in pairs. But I'm not sure how to do even and odd analysis on openEMS... is there a tutorial somewhere explaining how to do it?

**Comment by VolkerMuehlhaus** [comment, marked as answer]
Hi Leo,

this is a very valid technical question!

1. Waveguide ports can only be used for hollow waveguides. openEMS has no equivalent to HFSS wave ports.

2. Yes, I would use the CPW port (zero conductor thickness) for excitation, followed by your DUT which can be modelled with actual physical thickness. You can then place the measurement plane with voltage & current probes at the interface where your DUT starts.
I have created a modified version of the CPW_line.m tutorial example, and added rather large physical thickness of 50um to the DUT metal. According to transmission line calculator, the difference in Zline is ~7 Ohm, and indeed the results in openEMS change quite a bit.

![cpw_port_probes](https://github.com/thliebig/openEMS-Project/assets/47922981/202d0b90-d0ae-4822-9347-f69b8c546ec3)

[CPW_Line_thick.zip](https://github.com/thliebig/openEMS-Project/files/13829212/CPW_Line_thick.zip)

3. I use that method for modelling ports at GSG probe pads using my commercial FDTD solver, and it works there using two 100 ohm ports with identical port number and a voltage probe weight factor of 0.5, so I think you can build something equivalent in openEMS. The two (sub)ports must be excited at the same time, and I think you will need to add up voltages/currents in postprocessing.
However, I would stronly prefer method 2 above

While testing the thick metal DUT for method2, I noticed that the well matched line in example CPW_line.m has 58 Ohm line impedance using a (reliable) 2D EM cross section solver, when simulated with infinitely thin metal. So I have some doubts if the mesh in the openEMS example is really fine enough for accurate results, this needs to be verified in my opinion.

![zline_cpw_thin_thick](https://github.com/thliebig/openEMS-Project/assets/47922981/1396780e-4ed6-4856-a60e-42c6449bc42e)

**Comment by LeoGalloGomes** [comment]
@VolkerMuehlhaus to the rescue :)

1) Good to know that waveguide ports can only be used to excite hollow waveguides. One less option on the table, then.

2) Yes, I was worried about that divergence. I already saw a difference in Zc for microstrips between HFSS and openEMS for the same structure (a 50 ohm microstrip on sky130 BEOL). I saw an error of about -2% in openEMS. I think that's acceptable because I also saw deviations on the same order between HFSS and ADS' FEM/MoM. However a difference of 7 ohms isn't negligible... I think I need to run some tests.

3) I tried that approach. I had to assign 4 lumped ports and also number them from 1 to 4 (openEMS doesn't like different ports with the same numbering). But if I activated two "half" ports the results are completely nonsensical. But maybe I did make a mistake in post-processing. Could you give some insight on how to define the ports and add them? (I don't have that script anymore because I edited it to add the PEC sheet).

**Reply by VolkerMuehlhaus** [reply]
@LeoGalloGomes 

Attached is my attempt to creates two lumped ports per side. Note the 100 Ohm impedance because they are in parallel, and note the direction to excite/probe in-phase. 

I get reasonable results for Zin when using voltage weight factor 0.5, but don't get the correct S11/S21 so far. It might be that defining the ports as 100 Ohm but then evaluating the 50 Ohm parameters in calcport isn't valid. I'm running out of time for today, but maybe you can follow up and get the attached model to work.

Good luck!
Volker

[CPW_Line_thick_lumpedport.zip](https://github.com/thliebig/openEMS-Project/files/13830049/CPW_Line_thick_lumpedport.zip)

![cpw_results_lumpedport](https://github.com/thliebig/openEMS-Project/assets/47922981/08823c7b-af48-466f-b98e-d7a66e604301)

**Reply by VolkerMuehlhaus** [reply]
I finally found the trivial mistake: missing mesh line in x direction at the ports, leading to incorrect calculation of incident and reflected voltage at port 2. Now everything works as expected, also tested with open end for total reflection.

[CPW_Line_thick_lumpedport2.zip](https://github.com/thliebig/openEMS-Project/files/13834801/CPW_Line_thick_lumpedport2.zip)

**Comment by LeoGalloGomes** [comment]
I got as far as adding the lumped ports, but I made a mistake in post-processing. Thank you so much for the help again!
